synapse/storage/devices.py

```diff
@@ -557,6 +557,9 @@ def _prune_txn(txn):
             txn.execute(select_sql)
             rows = txn.fetchall()
 
+            if not rows:
+                return
```

Comment:
Should we chuck a `min(ts)` into the SELECT so that we can filter out destinations that don't have anything to delete in them? I'm slightly worried at the number of deletes that we are spamming here.